$MirOS: ports/mailnews/pine/patches/patch-pine_mailcap_c,v 1.1.7.1 2005/03/18 15:49:48 tg Exp $
$OpenBSD: patch-pine_mailcap_c,v 1.3 2003/10/05 21:42:29 jakob Exp $
--- pine/mailcap.c.orig	Wed Nov  3 20:10:55 2004
+++ pine/mailcap.c	Wed May  4 23:28:10 2005
@@ -993,14 +993,18 @@ mc_cmd_bldr(controlstring, type, subtype
 		     * have to put those outside of the single quotes.
 		     * (The parm+1000 nonsense is to protect against
 		     * malicious mail trying to overlow our buffer.)
+		     *
+		     * TCH - Change 2/8/1999
+		     * Also quote the ` slash to prevent execution of arbitrary code
 		     */
 		    for(p = parm; *p && p < parm+1000; p++){
-			if(*p == '\''){
+			if((*p == '\'') || (*p=='`')) {
 			    *to++ = '\'';  /* closing quote */
 			    *to++ = '\\';
-			    *to++ = '\'';  /* below will be opening quote */
-			}
-			*to++ = *p;
+			    *to++ = *p;		/* quoted character */
+			    *to++ = '\'';	/* opening quote */
+			} else
+			    *to++ = *p;
 		    }
 
 		    fs_give((void **) &parm);
